tests/model_fields/tests.py

```diff
@@ -271,6 +271,14 @@ class Meta:
 
         self.assertEqual(ConcreteReferent._meta.get_field('reference').related_model, Refered)
 
+    @test.skipUnlessDBFeature('can_defer_constraint_checks')
+    def test_defers_constraint_check(self):
+        b = Bar.objects.create(b='abc', a_id=5)
+        self.assertEqual(b.a_id, 5)
+        # Delete the object to avoid the error once constraints are
+        # checked.
+        b.delete()
```

Comment:
Is this a proper regression test? My understanding is that it will only fail (and cause a cascade a failures) if you omit `b.delete()`.